packages/website/src/components/RulesTable/index.tsx

```diff
@@ -122,78 +123,57 @@ export default function RulesTable({
 }: {
   extensionRules?: boolean;
 }): JSX.Element {
+  const [filters, changeFilter] = useRulesFilters(
+    extensionRules ? 'extension-rules' : 'supported-rules',
+  );
```

Comment:
[Cleanup] I'm not a big fan of hardcoding IDs on boolean logic. It tends to break over time, and makes it harder to understand where the IDs come from. How about giving the `RulesTable` component a mandatory `id` prop that's then passed directly to `useRulesFilters`?

Reply:
I agree about hardcoding is no boolean logic, but I feel that 

```<RulesTable id="extension-rules">```

might be a bit confusing given that the heading above receives `id` attribute from the markdown pipeline.

<img width="907" alt="image" src="https://user-images.githubusercontent.com/15332326/224913601-6a2789e7-fcd8-4f4d-bc1d-a0f8db681a85.png">

How about a prop `ruleset: "supported-rules" | "extension-rules"` that would subsume the boolean flag **and** be usable as a search param key?

Reply:
Yeah I like that a lot!

Reply:
I added it in https://github.com/typescript-eslint/typescript-eslint/pull/6568/commits/642031fe921885c740f552bc741865bffc53686a, but it seems there are indirect codecov changes that are failing. Do you think it's connected to my change?

Reply:
We can ignore codecov for these website PRs. We don't have great unit test coverage on it anyway. 🤷